src/components/bottom_bar/bottom_bar.styles.ts

```diff
@@ -23,34 +23,38 @@ const euiBottomBarAppear = keyframes`
   }
 `;
 
-export const euiBottomBarStyles = ({ euiTheme, colorMode }: UseEuiTheme) => ({
-  // Base
-  // Text color needs to be reapplied to properly scope the forced `colorMode`
-  euiBottomBar: css`
-    ${euiShadowFlat(euiTheme, undefined, colorMode)};
-    background: ${shade(euiTheme.colors.lightestShade, 0.5)};
-    color: ${euiTheme.colors.text};
-    ${euiCanAnimate} {
-      animation: ${euiBottomBarAppear} ${euiTheme.animation.slow}
-        ${euiTheme.animation.resistance};
-    }
-  `,
-  static: css``,
-  fixed: css`
-    z-index: ${Number(euiTheme.levels.header) - 2};
-  `,
-  sticky: css`
-    z-index: ${Number(euiTheme.levels.header) - 2};
-  `,
-  // Padding
-  s: css`
-    padding: ${euiTheme.size.s};
-  `,
-  m: css`
-    padding: ${euiTheme.size.base};
-  `,
-  l: css`
-    padding: ${euiTheme.size.l};
-  `,
-  none: '',
-});
+export const euiBottomBarStyles = (_euiTheme: UseEuiTheme) => {
+  const { euiTheme } = _euiTheme;
```

Comment:
This distinction feels somewhat confusing to me and I wish we were a bit more consistent on whether we pass the entire returned context or just `euiTheme` between various helpers (e.g., the typography fns all pass `euiTheme.euiTheme` and not the parent context, but this is also because basically every single `.styles.ts` file deconstructs `{ euiTheme }` automatically in its params.

Somewhat separately, I also wish we had a clearer naming convention for the context vs the actual theme data so that we don't end up essentially deconstructing `_euiTheme.euiTheme` 🙈 Maybe `themeContext.euiTheme` instead?...

@thompsongl, any thoughts here?

Reply:
Agreed with all those points. I wonder if it's just easiest to say, any mixin/function should accept the entirety of `UseEuiTheme` whether they only need the `.euiTheme` portion or not? It keeps them scalable in case of future needs to grab the `colorMode`.

As for naming we could just change to also ensure we're naming it as lowercase `useEuiTheme` so that it is clear.

```suggestion
export const euiBottomBarStyles = (useEuiTheme: UseEuiTheme) => {
  const { euiTheme } = useEuiTheme;
```

Reply:
++

Also I just realized we could simplify the code portion a bit too by deconstructing in the parameters.

```suggestion
export const euiBottomBarStyles = ({ euiTheme }: UseEuiTheme) => {

```

Reply:
I pushed a commit based on those latest comments. But I didn't touch the util (that one still confuses me a bit)

Reply:
I'd rather we not deconstruct the returned top-level values as `useEuiTheme` - in general the `use` prefix means something **very** specific to React (it's a hook + it's a function), neither of which apply to the variable being passed.

I'm also not a fan of `_theme` or `_euiTheme` with an underscore prefix, in general an underscore prefix means "this is internal" to me, _or_ it means "we're going to override this later with conditional logic" (in the case of our props), neither of which is actually true of the variable usage (it's not internal and it's not being overridden, it simply doesn't have a clear name).

This would be my suggestion for variable naming when passing the entire context:

```suggestion
export const euiBottomBarStyles = (euiThemeContext: UseEuiTheme) => {
  const { euiTheme } = euiThemeContext;
```

Reply:
> But I didn't touch the util

Looks like these weren't it anyway. We can do it in a separate PR(s) to update the other mixins.

Reply:
> But I didn't touch the util 

Can't really update the util unless we update all the mixins that use it (not just shadows). Do you want to do that in this PR?

Reply:
> This would be my suggestion for variable naming when passing the entire context:

Done and updated

> Do you want to do that in this PR?

**I** do not 😸

Reply:
> I do not 😸

😄 But are you ok if I do it and add a bunch to this PR?

Reply:
Or if you want to merge this I can do a follow up. Up to you!

Reply:
Yes, go for it. I'm out for the rest of the week.

Reply:
Hmm not sure I can make a generic function that works with mixins that do have required args and mixins that do not have required args.

```
const euiMixin(
  euiTheme: UseEuiTheme;
  required: PropertyThatIsRequired;
  optional?: {
    optionalKey1?: something;
    optionalKey2?: something;
  }
) => {}
```

vs

```
const euiMixin(
  euiTheme: UseEuiTheme;
  optional?: {
    optionalKey1?: something;
    optionalKey2?: something;
  }
) => {}
```

So a couple options:

* Only have the former and you're required to pass an empty arg when no properties are required: `euiScrollBarStyles(euiThemeContext, {}, {size: 20});`
* Move the required args and optional args into a single object:
```
const euiMixin(
  euiTheme: UseEuiTheme;
  options: {
    requiredKey1: something;
    optionalKey1?: something;
    optionalKey2?: something;
  }
) => {}
```
* Create a second version of `createStyleHookFromMixin` so we have one with required args and one without
* Scrap `createStyleHookFromMixin` entirely and manually create hooks. (We lose the type safety/consistency that it adds to mixins; we have to just be consistent when creating new mixins).

Reply:
Honestly I think my (opinionated) vote goes to

> Scrap createStyleHookFromMixin entirely and manually create hooks. (We lose the type safety/consistency that it adds to mixins; we have to just be consistent when creating new mixins).

I haven't used this helper personally despite having created a few mixins and I don't super see the need for it. 🤔 It's not really that tedious to manually write a 5-line hook next to a mixin.

If we do strongly want the helper still, then my next vote goes to `Move the required args and optional args into a single object`